server/handle_single_deployment.go

```
@@ -112,9 +113,32 @@ func (h *GETSingleDeploymentHandler) Exchange() (interface{}, int) {
 		return h.err(400, "Cannot deploy, current num instances set to zero, please update manifest.")
 	}
 
+	links := make(map[string]string)
+	baseURL := h.GDM.Defs.Clusters[did.Cluster].BaseURL
+	singularityRequestID := dep.DeployConfig.SingularityRequestID
+	singularityURL := makeSingularityURL(baseURL, singularityRequestID, did)
+	if len(singularityURL) > 0 {
+		links["SingularityURL"] = singularityURL
+	}
+
 	h.Body.Deployment = &dep
 
-	return h.ok(200, nil)
+	return h.ok(200, links)
+}
+
+func makeSingularityURL(baseURL string, singularityRequestID string, deploymentID sous.DeploymentID) string {
+	var err error
+	if len(singularityRequestID) == 0 {
+		singularityRequestID, err = singularity.MakeRequestID(deploymentID)
+		if err != nil {
+			return ""
```

I wonder if we should return this error to the user. Main reason being I can see people wanting to use this value in automated scripts, so just silently dropping it would be more confusing than, even, setting the URL to something like `fmt.Sprintf("Unable to determine SingularityRequest URL: %s", err)`.